Directory.Packages.props

```
@@ -23,7 +23,17 @@
     <PackageVersion Include="Microsoft.Extensions.DependencyInjection" Version="[3.1.0,)" />
     <PackageVersion Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="[3.1.0,)" />
     <PackageVersion Include="Microsoft.Extensions.Hosting.Abstractions" Version="[2.1.0,)" />
-    <PackageVersion Include="Microsoft.Extensions.Logging" Version="[3.1.0,)" />
+
+    <!--
+        Typically, the latest stable version of Microsoft.Extensions.Logging should be used here because:
+        1) Each major version bump will have some new API capabilities (e.g. .NET 6 introduced compile-time logging
+          source generation, .NET 8 introduced automatic event id generation).
+        2) Each minor version bump is normally security hotfixes or critical bug fixes.
+        3) Since version 3.1.0, the .NET runtime team provides extra backward compatibility guarantee to
```

```suggestion
        3) Since version 3.1.0, the .NET runtime team is holding a high bar for backward compatibility on
```

I don't think we guarantee back compatibility, but we do prioritize it pretty highly and I think we are largely successful. In terms of 'extra', I wouldn't say that M.E.Logging gets more back compat concern than any other library owned by the BCL team, but relative to the back-compat bar that ASP.NET team held during the 1.x and 2.x releases definitely a much higher bar is being held now.